Dockerfile

```diff
@@ -11,7 +11,7 @@ COPY go.mod go.mod
 COPY go.sum go.sum
 COPY providers/go.mod providers/go.mod
 COPY providers/go.sum providers/go.sum
-RUN go mod download
+COPY vendor/ vendor/
```

Comment:
I would put this in one line with go.mod, go.sum and vendor/, because if any of those change they should all change

structuring these in a good order will limit cache invalidation when iteratively building locally, which isn't strictly necesasry but is nice to have.

Reply:
(three lines also works, but should probably be before the other module, unless they're all using vendor via go.work and we're not copying that in)

vendor is only going to be used for the root module if we don't use go.work AFAIK

Reply:
I combined the lines that can be combined.  But doing the obvious thing doesn't work except with `# syntax=docker.io/docker/dockerfile:1.7-labs`, which seems like throwing good money after bad